Automattic/jetpack
# Jetpack: Improve UX when rendering the VideoPress player

The goal of this PR is improves the UX when the video player is rendered in the editor canvas, tackling two stations:

### The initial render of the block
Currently, when the block renders the height is unknown producing a height zero element. Once the block gets the video preview, it renders the sandbox (iFrame) to finally shows the player in its ready-to-use state.
This PR, gets the width and height of the video, provided by the preview object, to compute the video ratio to finally store it into a new block attribute.
With this ratio value, it is possible to set the block height either when the preview is not available. It's named like `temporaryHeight` that, as its name explains, it's used only when the preview is not ready. The result is shown below, comparing this implementation with the current one. Pay attention to the caption and the `<hr />`, observing the block height. It doesn't change with this suggestion.

https://user-images.githubusercontent.com/77539/177217288-4fba7e9b-9a8a-42b0-b132-8be1258761ff.mov

https://user-images.githubusercontent.com/77539/177217281-6abf27fa-bf3d-4895-b25d-fcb9558c2227.mov

### Fallback image
Taking advantage of the preview provided by the VideoPress infrastructure, the block stores the video thumbnail image in a new attribute.
This attribute is used to render the static image behind the Sandbox, providing a fallback image shown when the video preview is not ready, something that happens pretty often (editing video props, for instance).
The first video of the follow ones, shows how the block renders the fallback image when the user changes the video properties, producing a fetch to get the fresh preview.

https://user-images.githubusercontent.com/77539/177217143-856ff343-e9fe-4dec-b2b3-a151b7e90038.mov

The next one, shows how it doesn't show the video when the player attributes change (current implementation)

https://user-images.githubusercontent.com/77539/177217991-f636a192-6061-467e-a69c-1e82ec6d22b9.mov

Fixes #

#### Changes proposed in this Pull Request:

* Jetpack: Improve UX when rendering the VideoPress player:
  - Set a temporary height when the block is first rendered
  - Add an image fallback when the video preview is loading

#### Other information:


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Go to block editor
* Confirm the enhancements mentioned above, in the PR description.



---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1202514379845525
  - 0-as-1202514379845507
